flow/flow.h

```diff
@@ -50,6 +51,17 @@
 
 using namespace std::rel_ops;
 
+// Sets the seed for the deterministic random number generator on the current thread
+void setThreadLocalDeterministicRandomSeed(uint32_t seed);
+
+// Returns the random number generator that can be seeded. This generator should only 
+// be used in contexts where the choice to call it is deterministic.
+Reference<IRandom> deterministicRandom();
+
+// A random number generator that cannot be manually seeded and may be called in 
+// non-deterministic contexts.
+Reference<IRandom> nondeterministicRandom();
```

Comment:
Looking at this now, I'm not sure why I've declared these here and in IRandom.h. For a time I was considering moving them, but I don't think we actually need both. I'll clear this up tomorrow.